src/Scripting/Core/ObjectFormatter.cs

```diff
@@ -449,83 +471,179 @@ private string FormatPrimitive(object obj, bool quoteStrings, bool includeCodePo
                 return NullLiteral;
             }
 
-            if (obj is bool)
+            if (obj.GetType() == typeof(int))
```

Comment:
Maybe we could switch on ```GetPrimitiveSpecialType```?  Seems like that would make it easier to tell if we missed a case (plus, we wouldn't need two complete lists of primitive types).